src/journal/ObjectRecorder.cc

```diff
@@ -308,39 +330,62 @@ void ObjectRecorder::append_overflowed(uint64_t tid) {
 }
 
 void ObjectRecorder::send_appends(AppendBuffers *append_buffers) {
-  assert(m_lock.is_locked());
+  assert(m_lock->is_locked());
   assert(!append_buffers->empty());
 
-  uint64_t append_tid = m_append_tid++;
-  ldout(m_cct, 10) << __func__ << ": " << m_oid << " flushing journal tid="
-                   << append_tid << dendl;
-  C_AppendFlush *append_flush = new C_AppendFlush(this, append_tid);
-
-  librados::ObjectWriteOperation op;
-  client::guard_append(&op, m_soft_max_size);
-
   for (AppendBuffers::iterator it = append_buffers->begin();
        it != append_buffers->end(); ++it) {
     ldout(m_cct, 20) << __func__ << ": flushing " << *it->first
                      << dendl;
     it->first->set_flush_in_progress();
-    op.append(it->second);
-    op.set_op_flags2(CEPH_OSD_OP_FLAG_FADVISE_DONTNEED);
     m_size += it->second.length();
   }
-  m_in_flight_tids.insert(append_tid);
-  m_in_flight_appends[append_tid].swap(*append_buffers);
-
-  librados::AioCompletion *rados_completion =
-    librados::Rados::aio_create_completion(append_flush, NULL,
-                                           utils::rados_ctx_callback);
-  int r = m_ioctx.aio_operate(m_oid, rados_completion, &op);
-  assert(r == 0);
-  rados_completion->release();
+
+  m_pending_buffers.splice(m_pending_buffers.end(), *append_buffers,
+                           append_buffers->begin(), append_buffers->end());
+  if (!m_aio_scheduled) {
+    m_op_work_queue->queue(new FunctionContext(
+      [this] (int r) {
+        Mutex::Locker locker(*m_lock);
```

Comment:
Minor: this is probably large enough chunk of logic to deserve its own method instead of using a lamba.